ui/components/capture/src/main/java/com/google/jetpackcamera/ui/components/capture/ToggleSwitch.kt

```diff
@@ -0,0 +1,452 @@
+import androidx.compose.foundation.gestures.draggable
+import androidx.compose.foundation.gestures.rememberDraggableState
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.CameraAlt
+import androidx.compose.material.icons.filled.Videocam
+import androidx.compose.material.icons.outlined.CameraAlt
+import androidx.compose.material.icons.outlined.Videocam
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableFloatStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.CornerRadius
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.graphics.drawscope.DrawScope
+import androidx.compose.ui.graphics.drawscope.clipRect
+import androidx.compose.ui.graphics.drawscope.translate
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.graphics.vector.rememberVectorPainter
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.semantics.stateDescription
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.google.jetpackcamera.ui.components.capture.theme.PreviewPreviewTheme
+
+private const val DISABLED_ICON_ALPHA = 0.38f
+
+/**
+ * A custom Switch component built with Canvas, supporting drag gestures
+ * and custom icons for on/off states, styled for the JCA project.
+ *
+ * @param checked Whether the switch is currently checked (on).
+ * @param onCheckedChange Lambda called when the switch state changes.
+ * @param modifier Modifier for this component.
+ * @param enabled Whether the switch is interactive.
+ * @param trackColor The color of the track when unchecked.
+ * @param thumbColor The color of the thumb and the checked track.
+ * @param leftIcon The [ImageVector] (e.g., Google Symbol) for the 'off' state.
+ * @param rightIcon The [ImageVector] (e.g., Google Symbol) for the 'on' state.
+ * @param unSelectedIconColor The tint color for the 'off' icon.
+ * @param selectedIconColor The tint color for the 'on' icon.
+ */
+@Composable
+fun ToggleSwitch(
+    checked: Boolean,
+    onCheckedChange: (Boolean) -> Unit,
+    modifier: Modifier = Modifier,
+    onToggleWhenDisabled: () -> Unit = {},
+    enabled: Boolean = true,
+    trackColor: Color = MaterialTheme.colorScheme.surfaceContainerHighest,
+    thumbColor: Color = MaterialTheme.colorScheme.primary,
```

Comment:
According to the mock, this should be `MaterialTheme.colorScheme.primaryFixedDim`